fri/src/two_adic_pcs.rs

```diff
@@ -255,45 +269,56 @@ where
                 (mats, points)
             })
             .collect_vec();
-        let mats = mats_and_points
-            .iter()
-            .flat_map(|(mats, _)| mats)
-            .collect_vec();
 
-        let global_max_height = mats.iter().map(|m| m.height()).max().unwrap();
+        // Find the maximum height of a matrix in the batch. TODO: This should be given by the matrices in the first round right?
+        let global_max_height = mats_and_points
+            .iter()
+            .flat_map(|(mats, _)| mats.iter().map(|m| m.height()))
+            .max()
+            .unwrap();
         let log_global_max_height = log2_strict_usize(global_max_height);
 
         // For each unique opening point z, we will find the largest degree bound
         // for that point, and precompute 1/(z - X) for the largest subgroup (in bitrev order).
-        let inv_denoms = compute_inverse_denominators(&mats_and_points, Val::GENERATOR);
+        let inv_denoms = compute_inverse_denominators(&mats_and_points, Val::GENERATOR); // TODO: We have hard coded the shift to be Val::GENERATOR. Why is this okay?
```

Comment:
Good catch, I think we might have started with a baked-in rule that `GENERATOR` is the shift, then intended to generalize it but looks like that wasn't completed.